app/src/main/java/ch/epfl/sdp/map/GoogleMapApi.java

```diff
@@ -140,6 +135,13 @@ public Bitmap createSmallCircle(int color) {
     synchronized public void displayEntity(Displayable displayable) {
         activity.runOnUiThread(() -> {
             removeMarkers(displayable);
+            /*
+            if(displayable instanceof DisplayableWithIcon) {

```

Comment:
This was a nice idea, but I think it will be easier to move the case directly in the concerned class (for example, we will add a method displayEntity in the displayable interface and thus all the displayable will implement it. If we take the enemy, we will simply need to call "displaySmallIcon(displayable, "Enemy", R.drawable.enemy);" in the method)